### DISCUSSION TITLE
JW-Library issue with Zoom (not exactly related to Only-M, but we use it all together)

### DISCUSSION THREAD

**Discussion by robertkjr3d** [AntonyCorbett/OnlyM#452] [Q&A] [answered]
I cannot duplicate this problem every time.  Has anyone else experienced this?
Where JW-Library gets locked in a mode where anytime you even touch any Zoom window.  It knocks JW-Library's 2nd window off of projecting the Text (or whatever else it could be projecting...).  The only solution that I can find is to completely kill Zoom:   Then restart it.  If an AV brother decides to live with the situation during a midweek meeting, it is a real burden!

The setup is: Both Zoom's 1st and the (Dual-Screen) 2nd window are on the same (Main Screen...) Normally this works fine.  When sharing, Zoom's 2nd-Window disappears.  

But especially when my fellow brother is using the computer, somehow he is doing something that triggers this condition.  And I don't know how or what he is doing that causes it. (Since we have been back to meetings, it has not yet happened to me...)  I actually have the exact same setup at home.  In that through the entire pandemic.  I was sharing with Dual-Screens, for meetings.  I did notice the problem maybe once or twice.  So I do know this situation can happen.  

Even though the 2nd-Dual-Screen Zoom Window is on the main Screen for some reason, something triggers JW-Library to minimize when in this sick like "condition".  I don't know what it is.  But I wonder if a software solution is possible, or if I'm missing something.  The problem is: Restarting Zoom during the meeting is not desirable.

**Comment by AntonyCorbett** [comment, marked as answer]
This sounds like the problem that JwlMediaWin was designed to address. A note from the Wiki: 

"JwlMediaWin monitors the state of the JW Library media window and when it is launched (i.e. when the JW Library "Play video on second display" option is chosen), it modifies the window to make it more permanent. By default, the media window is quite fragile - if you move another window over it or launch an application on the same monitor it minimises. In normal operation of the audio-visual equipment at the Kingdom Hall this is not desirable and can lead to some disruption."

https://github.com/AntonyCorbett/JwlMediaWin

**Reply by robertkjr3d** [reply]
And what if we actually need to run Only-M (at the same time)?  Since we are a Chinese-Cong with a Lot-O-Englishites.  We run all are Pinyin-Songs through Only-M.  I personally use Only-M for a lot of stuff.  Because I turn on Subtitles and the Chinese-Natives love it! 

I'll test this JwlMediaWin.

**Reply by robertkjr3d** [reply]
Oooh... mesa likey!!!! That's it.  I can't break JW-Library's bond whileist that's running.  And at the same time I can run Only-M
![giphy](https://user-images.githubusercontent.com/3276551/172422718-1ea9e574-ee03-4647-aafe-a41c5ab66b05.gif)
!  I'll get this uploaded on the machine at the KH.

**Reply by AntonyCorbett** [reply]
OnlyM will display over the top of the JWL media window if configured correctly. There is an option in OnlyM to display its media window "On top" and this will show over the JWL media window. Alternatively, you can configure JwlMediaWin so that it doesn't force the JWL media window to remain "on top". This means other applications will always display over the top of it (but JwlMediaWin prevents this from disrupting the JWL media window).

**Reply by robertkjr3d** [reply]
The only issue that I can forsee.  Every once in a great while, we rarely have a Zoomist that needs to give a part.  In that case, I think we need to uncheck that 'Enable Fix'.  Sometimes its a lot to remember.

**Reply by AntonyCorbett** [reply]
Permanently turn off the "Keep on top" option in JwlMediaWin.

**Reply by robertkjr3d** [reply]
hmm... isn't that what made it unbreakable when I was just testing it?  I'll test it again.

**Reply by ColinAAvey** [reply]
We turn off the "Keep on top"
The way JwlMediaWin changes the JWLib media window makes a huge difference.
We also toggle the "Play video on second display" option in JWLib to show Zoom participant. This can be down without restarting JWLib.

![image](https://user-images.githubusercontent.com/46714776/172433589-19d9f5a7-3a71-4c44-9e44-35a76ed4d279.png)

**Reply by robertkjr3d** [reply]
On my home-setup (as well as the KH setup) I have forgotten to turn off the taskbar on the 2nd-screen.  (It doesn't make sense at home, but I should have done this when setting up the KH computer.)  What I'm getting at is:  When testing this without the setting 'Always on top' enabled:  Yes, the text is always there behind the foreground window.  However, the taskbar shows up with icons.  But this can be remedied, if I remove the taskbar.  So problem should be solved...  Thank you.
